Source/WebCore/dom/Element.cpp

```diff
@@ -2749,13 +2749,13 @@ bool Element::hasAutoTextDirectionState() const
 void Element::updateEffectiveTextDirection()
 {
     auto textDirectionState = elementTextDirectionState(*this);
-    updateEffectiveTextDirectionState(*this, textDirectionState);
+    updateEffectiveTextDirectionState(*this, textDirectionState, false);
 }
 
 void Element::dirAttributeChanged(const AtomString& newValue)
 {
     auto textDirectionState = parseTextDirectionState(newValue);
-    updateEffectiveTextDirectionState(*this, textDirectionState);
+    updateEffectiveTextDirectionState(*this, textDirectionState, true);
```

Comment:
Ditto.

Reply:
I changed the fix since the old one seems to be incorrect.